test/test_tensor_ops.py

```diff
@@ -0,0 +1,411 @@
+
+        # arbitrary object
+        yield SampleInput(object(), ref=RuntimeError('Could not infer dtype'))
+
+        # python data with heterogeneous types
+        yield SampleInput([0, 'torch'], ref=TypeError('invalid data type'))
+
+        # python data with self-referential lists
+        z = [0]
+        z += [z]
+        yield SampleInput(z, ref=TypeError('self-referential lists are incompatible'))
+        yield SampleInput([[1, 2], z], ref=TypeError('self-referential lists are incompatible'))
+
+        class BadCpuArrayInterface1:
+            __array_interface__ = {}
+
+        class BadCpuArrayInterface2:
+            __array_interface__ = []
+
+        class BadCpuArrayInterface3:
+            @property
+            def __array_interface__(self):
+                raise IndexError('bogus index error')
+
+        class BadCpuArrayInterface4:
+            def __array__(self):
+                return [1, 2]
+
+        class BadCpuArrayInterface5:
+            def __array__(self):
+                raise IndexError('bogus index error')
+
+        class BadCudaArrayInterface1:
+            __cuda_array_interface__ = {}
+
+        class BadCudaArrayInterface2:
+            __cuda_array_interface__ = []
+
+        class BadCudaArrayInterface3:
+            @property
+            def __cuda_array_interface__(self):
+                raise IndexError('bogus index error')
+
+        if device.type == 'cpu':
+            # bad CPU Array Interface implementations
+            yield SampleInput(BadCpuArrayInterface1(), ref=ValueError('Missing __array_interface__'))
+            yield SampleInput(BadCpuArrayInterface2(), ref=ValueError('Invalid __array_interface__'))
+            yield SampleInput(BadCpuArrayInterface3(), ref=IndexError('bogus index error'))
+            yield SampleInput(BadCpuArrayInterface4(), ref=ValueError('object __array__ method not producing an array'))
+            yield SampleInput(BadCpuArrayInterface5(), ref=IndexError('bogus index error'))
+        elif device.type == 'cuda':
+            # bad CUDA Array Interface implementations
+            yield SampleInput(BadCudaArrayInterface1(), ref=ValueError('Missing __cuda_array_interface__'))
+            yield SampleInput(BadCudaArrayInterface2(), ref=ValueError('Invalid __cuda_array_interface__'))
+            yield SampleInput(BadCudaArrayInterface3(), ref=IndexError('bogus index error'))
+
+
+tensor_op_db = [TensorCreationOpInfo('as_tensor',
+                                     op=torch.as_tensor,
+                                     sample_inputs_func=sample_inputs_tensor,
+                                     supports_out=False),
+                TensorCreationOpInfo('tensor',
+                                     op=torch.tensor,
+                                     sample_inputs_func=sample_inputs_tensor,
+                                     supports_out=False),
+                TensorCreationOpInfo('tensor_pinning',
```

Comment:
Typically an op with multiple OpInfos is really multiple operators (like torch.divide() actually being torch.true_divide(), torch.floor_divide(), and the implicit torch.trunc_divide()). I think we should try to stick to that pattern and not use new OpInfos as a mechanism to refine sample inputs.